### ISSUE TITLE
Cannot use RemoteApp Tool on Windows Server 2022 Datacenter Core

### ISSUE THREAD

**Issue by denisapain** [kimmknight/remoteapptool#105]
I tried to add a remoteapp, it crashed, and I cannot start it again, I tried reinstalling, using 6.0.0.0 portable version, maybe my entry is bugged, maybe I can remove it from the registry but I dont know how

**Comment by denisapain** [comment]
I was able to remove it from **HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Terminal Server\TSAppAllowList\Applications** but when I tried to add again it crashes and wont startup, I can't add it with the + button only from **File** '-> **New RemoteApp (advanced)...**

**Comment by denisapain** [comment]
The entry is in the registry, but I can't get the rdp file because remoteapp tool doesnt like having any remoteapp entries

**Comment by denisapain** [comment]
![image](https://github.com/kimmknight/remoteapptool/assets/126583428/bb62c156-bbec-40be-9983-9dd29f7075fb)
**this is what happens when I click the + button** (There is no error when it crashes and doesn't start up)

**Comment by MrBrianGale** [comment]
Pretty sure I can tell you the problem - it is because you are using the CORE version of Windows Server. As per Microsoft's documentation CORE doesn't include a GUI. As per the Microsoft documentation:
    there is no desktop in Server Core, by design
See - https://learn.microsoft.com/en-us/windows-server/administration/server-core/what-is-server-core

So if you REALLY installed the CORE edition and then did some hacks to get a GUI in place, chances are your OS is in an unsupported state and as such you are going to get errors. Since the error is coming up from a COM exception calling a standard windows function (show the file browser dialog), I am thinking that you have a broken install OR a misconfigured install of your OS.

The TL;DR version - CORE OS's aren't supported as they have no GUI and this is a GUI tool.

**Comment by denisapain** [comment]
I thought that was the issue, however the open file dialog works with
notepad and mspaint in core so not sure about that, I haven't done any
hacks I've rdped in and downloaded it with invoke-webrequest

However now I have reinstalled the desktop experience version and It does
work (expected)
